src/main_thread/api/public_api.ts

```
@@ -688,6 +688,24 @@ class Player extends EventEmitter<IPublicAPIEvent> {
     this._priv_lastAutoPlay = options.autoPlay;
   }
 
+  /**
+   * Destroy current media source and re-attach new one.
+   *
+   * This method is useful when the media source is corrupted and needs to be reloaded.
+   * In some of legacy CDM implementations (e.g. Tizen 3.0), they try to decrypt the segment with first loaded init segment
+   * When the assets starts with non-drm protected content and switch to drm protected segment
+   * CDM trying to use the unencrypted init segment to decrypt the encrypted segment user will see the content with green artifacts or black screen
+   *
+   * To fix this issue, we need to reload the media source to make sure the CDM is using the correct init segment
+   */
+  public reloadMediaSource(): void {
+    if (this._priv_contentInfos === null) {
+      throw new Error("API: No content is currently loaded.");
+    }
```

I wonder if we want this or to simply fail silently 🤔